src/rgw/rgw_common.h

```diff
@@ -1772,7 +1772,7 @@ struct req_init_state {
 #include "rgw_auth.h"
 
 /** Store all the state necessary to complete and respond to an HTTP request*/
-struct req_state {
+struct req_state : DoutPrefixProvider {
```

Comment:
do you see this as a mixin?

Reply:
not really - each class needs to provide its own `gen_prefix()`, so there isn't any common code to share between providers. inheritence seems like a good fit, because we'll want to be able to pass these as a `DoutPrefixProvider*` into generic code like `get_system_obj()` - and logging there would use a different prefix depending on whether it was called from a RGWOp, radosgw-admin, multisite, etc. does that make sense?